frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1763,9 +1763,6 @@ func pathRelativeToWorkingDir(s llb.State, p string, platform ocispecs.Platform)
 		return "", err
 	}
 
-	if len(p) == 0 {
```

Comment:
nit: removed so that empty directory `""` are handled properly, especially for non-root `WORKDIR`. That's where the regression happens.